### DISCUSSION TITLE
Customers with multiple bank account connection validation

### DISCUSSION THREAD

**Discussion by btjones-me** [codatio/codat-docs#250] [Q&A] [answered]
I'm working on an integration feature which ensures that a Customer has not set up separate connections to the same real-world bank account. While this is an unlikely scenario we have seen customers do this in the past. Is there any way to set up sandbox data such that two different Banking Account might have the same real-world `accountNumbers` values?

**Comment by dcoplowe** [comment, marked as answer]
Hi Benjamin,

When an SME is first asked to authorise the sharing of open banking data, you will need to share the white-label link URL. They will then follow the process, select their applicable bank i.e. login & then be redirected back to Codat's white-label flow. They have the option to click "Add more" as an SME might have multiple bank accounts and you would want a full view of their financial data. If they attempt to click the same bank, they are presented with an error message on the Codat side saying "You already have a bank connected. Make sure you aren't trying to connect the same bank account again." If they attempt to click through and add the same account the bank will handle this and it's managed on a bank by bank basis. Although, you would be able to easily identify multiple transactions if you called endpoint `/companies/{companyId}/data/banking-transactions` these would list out duplicated transactions in stacked order. I tested with my personal bank account, connected it twice before calling the endpoint and saw duplicate transactions, with the same IDs. If you suspect this to be an issue, I'd build a validation check your side in scenarios where more than 1 bank account is connected and complete a cross check for duplicate IDs via the above end point.

I hope that helps and let me know if you have any more questions.

Cheers,

David

**Reply by btjones-me** [reply]
Thanks very much David!
